crates/bevy_render/src/view/window/screenshot.rs

```diff
@@ -221,114 +427,178 @@ impl SpecializedRenderPipeline for ScreenshotToScreenPipeline {
+                let texture_format = gpu_image.texture_format;
+                let texture_view = gpu_image.texture_view.deref();
+                render_screenshot(
+                    encoder,
+                    prepared,
+                    pipelines,
+                    entity,
+                    width,
+                    height,
+                    texture_format,
+                    texture_view,
+                );
+            }
+            NormalizedRenderTarget::TextureView(texture_view) => {
+                let texture_view = manual_texture_views.get(texture_view).unwrap();
+                let width = texture_view.size.x;
+                let height = texture_view.size.y;
+                let texture_format = texture_view.format;
+                let texture_view = texture_view.texture_view.deref();
+                render_screenshot(
+                    encoder,
+                    prepared,
+                    pipelines,
+                    entity,
+                    width,
+                    height,
+                    texture_format,
+                    texture_view,
+                );
+            }
+        };
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+fn render_screenshot(
+    encoder: &mut CommandEncoder,
+    prepared: &RenderScreenshotsPrepared,
+    pipelines: &PipelineCache,
+    entity: &Entity,
+    width: u32,
+    height: u32,
+    texture_format: TextureFormat,
+    texture_view: &wgpu::TextureView,
+) {
+    if let Some(prepared_state) = &prepared.get(entity) {
+        encoder.copy_texture_to_buffer(
+            prepared_state.texture.as_image_copy(),
+            wgpu::ImageCopyBuffer {
+                buffer: &prepared_state.buffer,
+                layout: layout_data(width, height, texture_format),
+            },
+            Extent3d {
+                width,
+                height,
+                ..Default::default()
+            },
+        );
+
+        if let Some(pipeline) = pipelines.get_render_pipeline(prepared_state.pipeline_id) {
+            let mut pass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                label: Some("screenshot_to_screen_pass"),
+                color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                    view: texture_view,
+                    resolve_target: None,
+                    ops: wgpu::Operations {
+                        load: wgpu::LoadOp::Load,
+                        store: wgpu::StoreOp::Store,
+                    },
+                })],
+                depth_stencil_attachment: None,
+                timestamp_writes: None,
+                occlusion_query_set: None,
+            });
+            pass.set_pipeline(pipeline);
+            pass.set_bind_group(0, &prepared_state.bind_group, &[]);
+            pass.draw(0..3, 0..1);
         }
     }
 }
 
 pub(crate) fn collect_screenshots(world: &mut World) {
     let _span = info_span!("collect_screenshots");
```

Comment:
Not necessarily part of this PR but I think this span is never entered IIRC how tracing works. Maybe can be replaced with `#[intrument(level = "info", skip_all)]` on the fn?

Reply:
We don't seem to use the tracing proc macros in the engine for some reason, I'm not sure why. However, I'm pretty sure this should be guarded with `#[cfg(feature = "trace")]`, so thanks for pointing that out :p